_posts/2025-08-16-accessible-error-messages-in-forms.md

```diff
@@ -0,0 +1,158 @@
+
+Prefer **“required”** over an asterisk, because an asterisk assumes prior knowledge of the meaning of that icon. If you do use an asterisk, explain its meaning above the form.
+
+### Mark required or non-required fields?
+
+Which is better — indicate required or non-required fields? It depends. Your choice should be based on the form’s function, your CMS or plugin, and user research.
+
+An option could be: when most fields are required, mark the *optional* ones. If most fields are not required, mark the *required* ones.
+
+{: .example }
+For example: "Please complete all fields. If a field is not required, it will be labeled "(Optional)".
+
+Whatever you choose, be consistent within the form and across all forms on your website. Inform users above the form how field requirements are indicated.
+
+{: .note }
+Please note: for **screen readers** is necessary to always add the attributes [`aria-required`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-required) and [`aria-invalid`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-invalid) for required form field.
+
+## When should I check for errors?
+
+Websites can check for errors while typing, after the focus moves out of a field, or after the form is submitted. Which approach is most user-friendly?
+
+Checking during typing can be confusing. Imagine typing an email address and seeing “Invalid email address” after the first letter. Of course, it’s invalid — you haven’t finished yet! 
+
+That leaves two options: check after leaving a field, or after submission. These can be combined. For certain fields — like dates that must be in the future — immediate checking is helpful. But checking after submission is always necessary.
+
+{: .warning }
+HTML5 form validation is not accessible at this moment. The W3C summarizes the options in [Validating Input ](https://www.w3.org/WAI/tutorials/forms/validation/). "If your web browser supports HTML5, it will not allow you to submit the form without entering text into the input field. Instead, it will display a message that is generated by the web browser itself".  Write custom error messages and provide server side generated error messages. The W3C explains the options in [Validating Input ](https://www.w3.org/WAI/tutorials/forms/validation/).
+
+## How do I indicate that an answer is incomplete or filled out incorrectly?
+
+Use more than colour alone to indicate errors. A user with visual impairments or colour blindness may not notice a red outline.  Colour is helpful — but always include an error message in text too.
+
+* Clearly state whether a field is required.
+* Show errors with more than just colour or an icon.
+* Write out the error in text above the form field.
+* Add a summary of all errors above the form  helps the user quickly see what needs to be corrected.
```

Comment:
Either "Adding a summary of all errors..." or "Add a summary of all errors above the form to help the user..."